torch/distributed/pipelining/stage.py

```diff
@@ -935,6 +935,67 @@ def _validate_fwd_outputs(self, outputs: tuple[torch.Tensor, ...]):
             f"Stage {self.stage_index} forward outputs", expected_tensors_meta, outputs
         )
 
+    def _init_p2p_neighbors(self, ops: list[dist.P2POp]) -> list[dist.P2POp]:
+        """
+        Set up p2p communicators between previous and next stages
+        by sending a dummy tensor.
+        """
+
+        def to_global_rank(peer_rank: Optional[int]) -> int:
+            assert peer_rank is not None
+            return (
+                peer_rank
+                if self.group is None
+                else dist.get_global_rank(self.group, peer_rank)
+            )
+
+        next_stage_peer_rank = self.stage_index_to_group_rank.get(self.stage_index + 1)
+        prev_stage_peer_rank = self.stage_index_to_group_rank.get(self.stage_index - 1)
+
+        recv_tensor = torch.zeros(1, device=self.device)
+        send_tensor = torch.tensor(self.stage_index, device=self.device)
+        # forward
+        if not self.is_first:
+            ops.append(
+                dist.P2POp(
+                    dist.irecv,
+                    recv_tensor,
+                    to_global_rank(prev_stage_peer_rank),
```

Comment:
Hm did we not add group_rank arg to P2POp class? Could use that to avoid the helper above if so.

Reply:
Ah good point, saw that you added that. I will update. I think there are other places we also do this translation so I can update in those places too